packages/@aws-cdk/aws-appmesh/test/integ.mesh.ts

```diff
@@ -0,0 +1,160 @@
+import ec2 = require('@aws-cdk/aws-ec2');
+import cloudmap = require('@aws-cdk/aws-servicediscovery');
+import cdk = require('@aws-cdk/core');
+
+import * as appmesh from '../lib/';
+
+export const app = new cdk.App();
+
+const stack = new cdk.Stack(app, 'mesh-stack', {});
+
+const vpc = new ec2.Vpc(stack, 'vpc', {
+  natGateways: 1,
+});
+
+const namespace = new cloudmap.PrivateDnsNamespace(stack, 'test-namespace', {
+  vpc,
+  name: 'domain.local',
+});
+
+const mesh = new appmesh.Mesh(stack, 'mesh');
+const router = mesh.addVirtualRouter('router', {
+  portMappings: [
+    {
+      port: 8080,
+      protocol: appmesh.Protocol.HTTP,
+    },
+  ],
+});
+
+const virtualService = mesh.addVirtualService('service', {
+  virtualRouter: router,
+  virtualServiceName: `service1.domain.local`,
+});
+
+const node = mesh.addVirtualNode('node', {
+  hostname: 'node1',
+  namespace,
+  listener: {
+    portMappings: [
+      {
+        port: 8080,
+        protocol: appmesh.Protocol.HTTP,
+      },
+    ],
+    healthChecks: [
+      {
+        healthyThreshold: 3,
+        interval: cdk.Duration.seconds(5),
+        path: '/check-path',
+        port: 8080,
+        protocol: appmesh.Protocol.HTTP,
+        timeout: cdk.Duration.seconds(2),
+        unhealthyThreshold: 2,
+      },
+    ],
+  },
+  backends: [
+    {
+      virtualService,
+    },
+  ],
+});
+
+node.addBackend({
+  virtualService: new appmesh.VirtualService(stack, 'service-2', {
+    virtualServiceName: 'service2.domain.local',
+    mesh,
```

Comment:
It seems a little clunky to have to pass in mesh here...

Reply:
I suppose if you had called `mesh.addVirtualService()` you wouldn't have had to do that.